internal/website/content/howto/docstore/_index.md

```diff
@@ -1,14 +1,359 @@
 ---
 title: "Docstore"
 date: 2019-06-08T15:11:57-04:00
+lastmod: 2019-07-11T14:57:23-07:00
 draft: true
 showInSidenav: true
+toc: true
 ---
 
-Docstore provides an abstraction layer over common [document
-stores](https://en.wikipedia.org/wiki/Document-oriented_database) like Amazon
-DynamoDB, Google Cloud Firestore, and MongoDB. These guides show how to work
-with Docstore in the Go CDK.
-
+Docstore provides an abstraction layer over common
+[document stores](https://en.wikipedia.org/wiki/Document-oriented_database) like
+Amazon DynamoDB, Google Cloud Firestore, and MongoDB. These guides show how to
+work with Docstore in the Go CDK.
 
 <!--more-->
+
+## Opening a Collection {#opening}
+
+The first step in using Docstore is connecting to your document store provider.
+Every document store provider is a little different, but the Go CDK lets you
+interact with all of them using the [`*docstore.Collection`][] type.
+
+While every provider has the concept of a primary key that uniquely
+distinguishes a document in a collection, each one specifies that key in its own
+way. To be portable, Docstore requires that the key be part of the document's
+contents. When you open a collection using one of the functions described here,
+you specify how to find the provider's primary key in the document.
+
+[`*docstore.Collection`]: https://godoc.org/gocloud.dev/docstore#Collection
+
+### Constructors versus URL openers
+
+The easiest way to open a collection is using [`docstore.OpenCollection`][] and
+a URL pointing to the topic, making sure you ["blank import"][] the driver
+package to link it in. See [Concepts: URLs][] for more details. If you need
+fine-grained control over the connection settings, you can call the constructor
+function in the driver package directly (like `mongodocstore.OpenCollection`).
+This guide will show how to use both forms for each document store provider.
+
+[`docstore.OpenCollection`]:
+https://godoc.org/gocloud.dev/docstore#OpenCollection
+["blank import"]: https://golang.org/doc/effective_go.html#blank_import
+
+\[Concepts: URLs]: {{< ref "/concepts/urls.md" >}}
+
+### DynamoDB {#dynamodb}
+
+The [`awsdynamodb`](https://godoc.org/gocloud.dev/docstore/awsdynamodb) package
+supports [Amazon DynamoDB](https://aws.amazon.com/dynamodb). A Docstore
+collection corresponds to a DynamoDB table.
+
+DynamoDB URLs provide the table, partition key field and optionally the sort key
+field for the collection.
+
+{{< goexample
+"gocloud.dev/docstore/awsdynamodb.Example_openCollectionFromURL" >}}
+
+Full details about acceptable URLs can be found under the API reference for
+[`awsdynamodb.URLOpener`][].
+
```

Comment:
These should be hosted under a parent title "Supported Docstore Services" or "Supported Services". Do you have a snapshot of the TOC from local hugo run? Compare that to the one for blob.

Reply:
Done.